### PR TITLE
WIP: add DeleteStrategy to generic registry stores

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/11980

### CODE REVIEW

**Comment by mfojtik** [comment]
[test]

File: pkg/image/registry/imagestream/etcd/etcd.go [hunk 1]
```diff
@@ -63,13 +64,15 @@ func NewREST(optsGetter restoptions.Getter, defaultRegistry api.DefaultRegistry,
 	statusStore.Decorator = nil
 	statusStore.CreateStrategy = nil
 	statusStore.UpdateStrategy = statusStrategy
+	statusStore.DeleteStrategy = nil
```
Comment (liggitt) [inline pkg/image/registry/imagestream/etcd/etcd.go]: Will a nil here cause issues?

Comment (mfojtik) [inline pkg/image/registry/imagestream/etcd/etcd.go]: @ncdc ideas?

Comment (ncdc) [inline pkg/image/registry/imagestream/etcd/etcd.go]: Can't have nil once we drop https://github.com/openshift/kubernetes/commit/0e612f0ab1227466f701991d03a627b5bba2ffe7

**Comment by mfojtik** [comment]
@liggitt @ncdc fixed.

**Comment by mfojtik** [comment]
@pweil- it seems like SCC need to have DeleteStrategy as well upstream.

**Comment by pweil-** [comment]
> @pweil- it seems like SCC need to have DeleteStrategy as well upstream.

was added for PSP in https://github.com/kubernetes/kubernetes/commit/31b425b3a189fc4ceee377038ff4337a6dca481c#diff-e875a75f877e0df1e0736fbe106700a7R63.  SCC is a carry

**Pull request closed without merging**